### ISSUE TITLE
[Enhancement Suggestion] auto save typed queries and reload them at start

### ISSUE THREAD

**Issue by jonathanantoine** [sqlitebrowser/sqlitebrowser#1140]
### Details for the issue

Thanks a lot for this tool !

It would be nice if typed queries would be saved at the exit of the app by db.
SqliteBrower would then reload then at the next launch for the same db. 

Something like Notepad++ is doing.

Regards.

#### I'm opening this issue because:
- [X ] DB4S needs a feature

**Comment by chrisjlocke** [comment]
Although not 'auto saved', you can save queries for a database in a project file. Click 'File' then 'Save Project'.

**Comment by jonathanantoine** [comment]
You are right and my initial ask for enhancement is not detailed enough. 

It would be nice if you dont even have to save them on disk.

**Comment by justinclift** [comment]
Hmmm, it's probably do-able but not sure when we'd get to it.

Thinking aloud here, we'd probably want to make use of our existing project file saving and loading code.  So, we'd add some kind of automatic thing to that, so that when a SQL query is run in the Execute tab a project file for the database is automatically created (or updated) in some kind of DB4S private caching area.

Then when loading a database, DB4S checks if there's a matching (automatic) project file and if there is then loads it.

The private caching area we already have the start of, as that's where we save databases people download from the DBHub server.  So this would probably extend that caching area out in some way.

**Comment by jonathanantoine** [comment]
Thanks a lot. This would be very very very appreciated!

**Comment by chrisjlocke** [comment]
Just another 'thought out loud', but this would break the 'portable' versions of DB4S, but should be easily fixed.  Noting it here for a reminder to me to bear in mind.

**Comment by justinclift** [comment]
Good point.  Any idea how to do things so the portable version would still work?  Maybe have the cache directory be optional, or perhaps somehow point it to the portable device?

**Comment by mgrojo** [comment]
We could store automatically a project file for each one of the recent database files. Each automatic project file might be stored as an entry in our settings, supposing that QSettings allows saving such long entries. In this way the user just opens a db file from the recent files and we automatically load the associated project entry in the settings. The disadvantage is that we would only store at most 5 entries, but then we'd avoid issues with an unlimited number of automatically saved files.

I don't know how the portable version is working, but if the saving of settings is already addressed, then this implementation would just work.
